src/reading.rs

```diff
@@ -73,6 +73,72 @@ impl ReadingOptions {
     }
 }
 
+/// Convenience builder to create a reader directly from file sources
```

Comment:
Would be good to have a small example docstring so that we have both a documentation and a small test